esp/services/ws_logaccess/WsLogAccessService.cpp

```diff
@@ -0,0 +1,147 @@
+    }
+    catch (IException * e)
+    {
+        StringBuffer msg;
+        e->errorMessage(msg);
+        LOG(MCoperatorWarning,"WsLogAccessService could not load remote log access plug-in: %s", msg.str());
+        e->Release();
+    }
+}
+
+LogAccessTimeRange requestedRangeToLARange(IConstTimeRange & reqrange)
+{
+    struct LogAccessTimeRange range;
+
+    range.setStart(reqrange.getStartDate());
+    range.setEnd(reqrange.getEndDate());
+
+    return range;
+}
+
+bool Cws_logaccessEx::onGetLogs(IEspContext &context, IEspGetLogsRequest &req, IEspGetLogsResponse & resp)
+{
+    const char * message = "success";
+    bool success = true;
+
+    if (m_remoteLogAccessor)
+    {
+        CLogAccessType searchByCategory = req.getLogCategory();
+        const char * searchByValue = req.getSearchByValue();
+        if (searchByCategory != CLogAccessType_All && isEmptyString(searchByValue))
+        {
+            message = "WsLogAccess::onGetLogs: Must provide log category";
+            success = false;
+        }
+        else
+        {
+            LogAccessConditions logFetchOptions;
+            switch (searchByCategory)
+            {
+                case CLogAccessType_All:
+                    logFetchOptions.setFilter(getWildCardLogAccessFilter());
+                    break;
+                case CLogAccessType_ByWUID:
+                    logFetchOptions.setFilter(getJobIDLogAccessFilter(searchByValue));
+                    break;
+                case CLogAccessType_ByComponent:
+                    logFetchOptions.setFilter(getComponentLogAccessFilter(searchByValue));
+                    break;
+                case CLogAccessType_ByLogType:
+                {
+                    LogMsgClass logType = LogMsgClass(LogMsgClassFromAbbrev(searchByValue));
+                    if (logType == 0 )
+                        throw makeStringExceptionV(-1, "Invalid Log Type 3-letter code encountered: '%s' - Available values: 'DIS,ERR,WRN,INF,PRO,MET'", searchByValue);
+                    logFetchOptions.setFilter(getClassLogAccessFilter(logType));
+                    break;
+                }
+                case CLogAccessType_ByTargetAudience:
+                {
+                    MessageAudience targetAud = MessageAudience(LogMsgAudFromAbbrev(searchByValue));
+                    if (targetAud == 0 || targetAud == MSGAUD_all)
+                        throw makeStringExceptionV(-1, "Invalid Target Audience 3-letter code encountered: '%s' - Available values: 'OPR,USR,PRO,ADT'", searchByValue);
+
+                    logFetchOptions.setFilter(getAudienceLogAccessFilter(targetAud));
+                    break;
+                }
+                case LogAccessType_Undefined:
+                    throw makeStringException(-1, "Invalid remote log access request type");
+                default:
+                    break;
+            }
+
+            struct LogAccessTimeRange range = requestedRangeToLARange(req.getRange());
```

Comment:
minor/style: unusual to qualify a struct (or a class) on a declaration like this.